internal/brokers/examplebroker.go

```diff
@@ -273,71 +296,107 @@ func (b *exampleBroker) IsAuthorized(ctx context.Context, sessionID, authenticat
 		}
 	}
 
-	authDeniedResp := "denied"
+	// Handle authentication in a separate routine to allow for cancellation
+	cleanup, err := b.newIsAuthorizedCall(ctx, sessionID)
```

Comment:
As IsAuthorized() is now more readable, I would not abstract this in a separate call, but just before, to make it more in the flow.

Reply:
As you wish. I separated it because there's some mutex handling in the calls and it gets a bit annoying to have it all the time in the "main" flow.